platform/entanglement/src/capabilities/LocationCapability.js

```diff
@@ -0,0 +1,77 @@
+/*global define */
+
+define(
+    function () {
+        "use strict";
+
+        function LocationCapability(domainObject) {
+            this.domainObject = domainObject;
+            return this;
+        }
+
+        /**
+         * Persist the current location of the current domain object as it's
+         * primary location.  Returns a promise.
+         */
+        LocationCapability.prototype.persistLocation = function () {
+            return this.domainObject.useCapability(
+                'mutation',
+                function (model) {
+                    model.location = this.getLocation();
+                }.bind(this)
+            ).then(function () {
+                return this.domainObject
+                    .getCapability('persistence')
+                    .persist();
+            }.bind(this));
+        };
+
+        /**
+         * Return the current location of the current domain object.  Only
+         * valid for domain objects that have a context capability.
+         */
+        LocationCapability.prototype.getLocation = function () {
+            var context = this.domainObject.getCapability("context"),
+                pathObjects,
+                pathIds;
+
+            if (!context) {
+                return '';
+            }
+
+            return context.getParent().getId();
+        };
+
+        /**
+         * Returns true if the domainObject is a link, false if it's an
+         * original.
+         */
+        LocationCapability.prototype.isLink = function () {
+            if (this.domainObject.getId() === "mine") {
```

Comment:
Am I right that this check is because My Items always shows up as a link otherwise? I think we can fix that in the definition of My Items (give it `location: "ROOT"`)